Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule_test.py

```diff
@@ -4228,3 +4228,69 @@ def test_request_for_bin_file_via_demisto_call(mocker, allow_bin_response):
         assert res == test_bin_data
     except DemistoException as e:
         assert f'{ALLOW_BIN_CONTENT_RESPONSE_SERVER_VERSION}-{ALLOW_BIN_CONTENT_RESPONSE_BUILD_NUM}' in str(e)
+
+
+def test_terminate_process_command(mocker):
+    """
+    Given:
+        - An XSIAM machine with a build version that supports demisto._apiCall() with RBAC validations.
+        - instance_id_1
+        - instance_id_2
+        - agent_id
+    When:
+        - Calling the terminate_process_command method.
+    Then:
+        - case 1 - Make sure the response are as expected (action_id).
+    """
+    from CoreIRApiModule import CoreClient, terminate_process_command
+    client = CoreClient(
+        base_url=f'{Core_URL}/public_api/v1', headers={},
+    )
+
+    mocker.patch("CoreIRApiModule.FORWARD_USER_RUN_RBAC", new=True)
+    mocker.patch.object(demisto, "_apiCall", side_effect=[
+        {'name': '/api/webapp/public_api/v1/endpoints/terminate_process',
+         'status': 200,
+         'data': json.dumps({'group_action_id': 1})},
+        {'name': '/api/webapp/public_api/v1/endpoints/terminate_process',
+         'status': 200,
+         'data': json.dumps({'group_action_id': 2})}
+    ]
+    )
+
+    result = terminate_process_command(client=client, args={'agent_id': '1', 'instance_id': ['instance_id_1', 'instance_id_2']})
+    assert result.readable_output == '### Action terminate process created on instance ids: instance_id_1, instance_id_2\n|action_id|\n|---|\n| 1 |\n| 2 |\n'
+    assert result.raw_response == [{'action_id': 1}, {'action_id': 2}]
+
+
+def test_terminate_causality_command(mocker):
+    """
+    Given:
+        - An XSIAM machine with a build version that supports demisto._apiCall() with RBAC validations.
+        - causality_id
+        - agent_id
+    When:
+        - Calling the terminate_causality_command method.
+    Then:
+        - case 1 - Make sure the response are as expected (action_id).
+    """
+    from CoreIRApiModule import CoreClient, terminate_causality_command
+    client = CoreClient(
+        base_url=f'{Core_URL}/public_api/v1', headers={},
+    )
+
+    mocker.patch("CoreIRApiModule.FORWARD_USER_RUN_RBAC", new=True)
+    mocker.patch.object(demisto, "_apiCall", side_effect=[
+        {'name': '/api/webapp/public_api/v1/endpoints/terminate_causality',
+         'status': 200,
+         'data': json.dumps({'group_action_id': 1})},
+        {'name': '/api/webapp/public_api/v1/endpoints/terminate_causality',
+         'status': 200,
+         'data': json.dumps({'group_action_id': 2})}
+    ]
+    )
+
+    result = terminate_causality_command(client=client, args={'agent_id': '1', 'causality_id': [
+                                         'causality_id_1', 'causality_id_2']})
+    assert result.readable_output == '### Action terminate causality created on causality_id_1,causality_id_2\n|action_id|\n|---|\n| 1 |\n| 2 |\n'
+    assert result.raw_response == [{'action_id': 1}, {'action_id': 2}]
```

Comment:
Lets add a test to test the change you made regarding raise DemistoException(f"The command {cmd} failed. Received status {result}")